### PR TITLE
grc: Fix YAML hier blocks (next)

### PR DESCRIPTION
**Pull request by haakov**
Switched to YAML for hierarchical blocks

### CODE REVIEW

File: grc/core/generator/top_block.py [hunk 1]
```diff
@@ -234,15 +234,20 @@ def make_port_sig(port):
...
+        for connection in virtual_source_connections:
             sink = connection.sink_port
             for source in connection.source_port.resolve_virtual_source():
                 resolved = connection_factory(fg.orignal_flowgraph, source, sink)
                 connections.append(resolved)
             # Remove the virtual connection
             connections.remove(connection)
 
+        for connection in virtual_sink_connections:
+            connections.remove(connection)
```
Comment (haakov) [inline grc/core/generator/top_block.py]: This prevents lines like `self.connect((self.analog_sig_source_x_0, 0), (self.virtual_sink_0, 0))` from appearing in the Python files. Attempting to use connect() with a virtual source/sink creates an `AttributeError: 'top_block_sptr' object has no attribute 'virtual_sink_0'`.

**Comment by marcusmueller** [comment]
so, following the discussion on #1886, we should probably be merging this on next, right? Or did we already?

**Comment by haakov** [comment]
This hasn’t been merged on next yet. I actually don’t think this PR has been touched since I created it in November, so because of that I’d like a couple of days to make sure it works if that’s okay. I’ll notify you when it’s been tested!

**Comment by haakov** [comment]
Okay so the first commit looks fine, but the one regarding virtual connections does not work with `next`. I'll have a look at it.

**Comment by haakov** [comment]
@marcusmueller It looks like the virtual connections are going to take some time to figure out, so I removed the commit that didn't work and squashed the others. This way you can merge this PR whenever you'd like and don't have to wait for me to figure out how to fix the virtual connections :)

**Comment by marcusmueller** [comment]
Thanks for the incredible patience! And, even more so, the incredible work :)

**Pull request merged**